File: src/OrdersStatsOrderDTO.cpp
#include <map>
#include <cstdlib>
#include <glib-object.h>
#include <json-glib/json-glib.h>
#include "Helpers.h"


#include "OrdersStatsOrderDTO.h"

using namespace std;
using namespace Tizen::ArtikCloud;

OrdersStatsOrderDTO::OrdersStatsOrderDTO()
{
	//__init();
}

OrdersStatsOrderDTO::~OrdersStatsOrderDTO()
{
	//__cleanup();
}

void
OrdersStatsOrderDTO::__init()
{
	//id = long(0);
	//creationDate = null;
	//statusUpdateDate = null;
	//status = new OrderStatsStatusType();
	//partnerOrderId = std::string();
	//paymentType = new OrdersStatsOrderPaymentType();
	//fake = bool(false);
	//deliveryRegion = new OrdersStatsDeliveryRegionDTO();
	//new std::list()std::list> items;
	//new std::list()std::list> initialItems;
	//new std::list()std::list> payments;
	//new std::list()std::list> commissions;
}

void
OrdersStatsOrderDTO::__cleanup()
{
	//if(id != NULL) {
	//
	//delete id;
	//id = NULL;
	//}
	//if(creationDate != NULL) {
	//
	//delete creationDate;
	//creationDate = NULL;
	//}
	//if(statusUpdateDate != NULL) {
	//
	//delete statusUpdateDate;
	//statusUpdateDate = NULL;
	//}
	//if(status != NULL) {
	//
	//delete status;
	//status = NULL;
	//}
	//if(partnerOrderId != NULL) {
	//
	//delete partnerOrderId;
	//partnerOrderId = NULL;
	//}
	//if(paymentType != NULL) {
	//
	//delete paymentType;
	//paymentType = NULL;
	//}
	//if(fake != NULL) {
	//
	//delete fake;
	//fake = NULL;
	//}
	//if(deliveryRegion != NULL) {
	//
	//delete deliveryRegion;
	//deliveryRegion = NULL;
	//}
	//if(items != NULL) {
	//items.RemoveAll(true);
	//delete items;
	//items = NULL;
	//}
	//if(initialItems != NULL) {
	//initialItems.RemoveAll(true);
	//delete initialItems;
	//initialItems = NULL;
	//}
	//if(payments != NULL) {
	//payments.RemoveAll(true);
	//delete payments;
	//payments = NULL;
	//}
	//if(commissions != NULL) {
	//commissions.RemoveAll(true);
	//delete commissions;
	//commissions = NULL;
	//}
	//
}

void
OrdersStatsOrderDTO::fromJson(char* jsonStr)
{
	JsonObject *pJsonObject = json_node_get_object(json_from_string(jsonStr,NULL));
	JsonNode *node;
	const gchar *idKey = "id";
	node = json_object_get_member(pJsonObject, idKey);
	if (node !=NULL) {
	

		if (isprimitive("long long")) {
			jsonToValue(&id, node, "long long", "");
		} else {
			
		}
	}
	const gchar *creationDateKey = "creationDate";
	node = json_object_get_member(pJsonObject, creationDateKey);
	if (node !=NULL) {
	

		if (isprimitive("Date")) {
			jsonToValue(&creationDate, node, "Date", "Date");
		} else {
			
		}
	}
	const gchar *statusUpdateDateKey = "statusUpdateDate";
	node = json_object_get_member(pJsonObject, statusUpdateDateKey);
	if (node !=NULL) {
	

		if (isprimitive("std::string")) {
			jsonToValue(&statusUpdateDate, node, "std::string", "");
		} else {
			
		}
	}
	const gchar *statusKey = "status";
	node = json_object_get_member(pJsonObject, statusKey);
	if (node !=NULL) {
	

		if (isprimitive("OrderStatsStatusType")) {
			jsonToValue(&status, node, "OrderStatsStatusType", "OrderStatsStatusType");
		} else {
			
			OrderStatsStatusType* obj = static_cast<OrderStatsStatusType*> (&status);
			obj->fromJson(json_to_string(node, false));
			
		}
	}
	const gchar *partnerOrderIdKey = "partnerOrderId";
	node = json_object_get_member(pJsonObject, partnerOrderIdKey);
	if (node !=NULL) {
	

		if (isprimitive("std::string")) {
			jsonToValue(&partnerOrderId, node, "std::string", "");
		} else {
			
		}
	}
	const gchar *paymentTypeKey = "paymentType";
	node = json_object_get_member(pJsonObject, paymentTypeKey);
	if (node !=NULL) {
	

		if (isprimitive("OrdersStatsOrderPaymentType")) {
			jsonToValue(&paymentType, node, "OrdersStatsOrderPaymentType", "OrdersStatsOrderPaymentType");
		} else {
			
			OrdersStatsOrderPaymentType* obj = static_cast<OrdersStatsOrderPaymentType*> (&paymentType);
			obj->fromJson(json_to_string(node, false));
			
		}
	}
	const gchar *fakeKey = "fake";
	node = json_object_get_member(pJsonObject, fakeKey);
	if (node !=NULL) {
	

		if (isprimitive("bool")) {
			jsonToValue(&fake, node, "bool", "");
		} else {
			
		}
	}
	const gchar *deliveryRegionKey = "deliveryRegion";
	node = json_object_get_member(pJsonObject, deliveryRegionKey);
	if (node !=NULL) {
	

		if (isprimitive("OrdersStatsDeliveryRegionDTO")) {
			jsonToValue(&deliveryRegion, node, "OrdersStatsDeliveryRegionDTO", "OrdersStatsDeliveryRegionDTO");
		} else {
			
			OrdersStatsDeliveryRegionDTO* obj = static_cast<OrdersStatsDeliveryRegionDTO*> (&deliveryRegion);
			obj->fromJson(json_to_string(node, false));
			
		}
	}
	const gchar *itemsKey = "items";
	node = json_object_get_member(pJsonObject, itemsKey);
	if (node !=NULL) {
	
		{
			JsonArray* arr = json_node_get_array(node);
			JsonNode*  temp_json;
			list<OrdersStatsItemDTO> new_list;
			OrdersStatsItemDTO inst;
			for (guint i=0;i<json_array_get_length(arr);i++) {
				temp_json = json_array_get_element(arr,i);
				if (isprimitive("OrdersStatsItemDTO")) {
					jsonToValue(&inst, temp_json, "OrdersStatsItemDTO", "");
				} else {
					
					inst.fromJson(json_to_string(temp_json, false));
					
				}
				new_list.push_back(inst);
			}
			items = new_list;
		}
		
	}
	const gchar *initialItemsKey = "initialItems";
	node = json_object_get_member(pJsonObject, initialItemsKey);
	if (node !=NULL) {
	
		{
			JsonArray* arr = json_node_get_array(node);
			JsonNode*  temp_json;
			list<OrdersStatsItemDTO> new_list;
			OrdersStatsItemDTO inst;
			for (guint i=0;i<json_array_get_length(arr);i++) {
				temp_json = json_array_get_element(arr,i);
				if (isprimitive("OrdersStatsItemDTO")) {
					jsonToValue(&inst, temp_json, "OrdersStatsItemDTO", "");
				} else {
					
					inst.fromJson(json_to_string(temp_json, false));
					
				}
				new_list.push_back(inst);
			}
			initialItems = new_list;
		}
		
	}
	const gchar *paymentsKey = "payments";
	node = json_object_get_member(pJsonObject, paymentsKey);
	if (node !=NULL) {
	
		{
			JsonArray* arr = json_node_get_array(node);
			JsonNode*  temp_json;
			list<OrdersStatsPaymentDTO> new_list;
			OrdersStatsPaymentDTO inst;
			for (guint i=0;i<json_array_get_length(arr);i++) {
				temp_json = json_array_get_element(arr,i);
				if (isprimitive("OrdersStatsPaymentDTO")) {
					jsonToValue(&inst, temp_json, "OrdersStatsPaymentDTO", "");
				} else {
					
					inst.fromJson(json_to_string(temp_json, false));
					
				}
				new_list.push_back(inst);
			}
			payments = new_list;
		}
		
	}
	const gchar *commissionsKey = "commissions";
	node = json_object_get_member(pJsonObject, commissionsKey);
	if (node !=NULL) {
	
		{
			JsonArray* arr = json_node_get_array(node);
			JsonNode*  temp_json;
			list<OrdersStatsCommissionDTO> new_list;
			OrdersStatsCommissionDTO inst;
			for (guint i=0;i<json_array_get_length(arr);i++) {
				temp_json = json_array_get_element(arr,i);
				if (isprimitive("OrdersStatsCommissionDTO")) {
					jsonToValue(&inst, temp_json, "OrdersStatsCommissionDTO", "");
				} else {
					
					inst.fromJson(json_to_string(temp_json, false));
					
				}
				new_list.push_back(inst);
			}
			commissions = new_list;
		}
		
	}
}

OrdersStatsOrderDTO::OrdersStatsOrderDTO(char* json)
{
	this->fromJson(json);
}

char*
OrdersStatsOrderDTO::toJson()
{
	JsonObject *pJsonObject = json_object_new();
	JsonNode *node;
	if (isprimitive("long long")) {
		long long obj = getId();
		node = converttoJson(&obj, "long long", "");
	}
	else {
		
	}
	const gchar *idKey = "id";
	json_object_set_member(pJsonObject, idKey, node);
	if (isprimitive("Date")) {
		Date obj = getCreationDate();
		node = converttoJson(&obj, "Date", "");
	}
	else {
		
	}
	const gchar *creationDateKey = "creationDate";
	json_object_set_member(pJsonObject, creationDateKey, node);
	if (isprimitive("std::string")) {
		std::string obj = getStatusUpdateDate();
		node = converttoJson(&obj, "std::string", "");
	}
	else {
		
	}
	const gchar *statusUpdateDateKey = "statusUpdateDate";
	json_object_set_member(pJsonObject, statusUpdateDateKey, node);
	if (isprimitive("OrderStatsStatusType")) {
		OrderStatsStatusType obj = getStatus();
		node = converttoJson(&obj, "OrderStatsStatusType", "");
	}
	else {
		
		OrderStatsStatusType obj = static_cast<OrderStatsStatusType> (getStatus());
		GError *mygerror;
		mygerror = NULL;
		node = json_from_string(obj.toJson(), &mygerror);
		
	}
	const gchar *statusKey = "status";
	json_object_set_member(pJsonObject, statusKey, node);
	if (isprimitive("std::string")) {
		std::string obj = getPartnerOrderId();
		node = converttoJson(&obj, "std::string", "");
	}
	else {
		
	}
	const gchar *partnerOrderIdKey = "partnerOrderId";
	json_object_set_member(pJsonObject, partnerOrderIdKey, node);
	if (isprimitive("OrdersStatsOrderPaymentType")) {
		OrdersStatsOrderPaymentType obj = getPaymentType();
		node = converttoJson(&obj, "OrdersStatsOrderPaymentType", "");
	}
	else {
		
		OrdersStatsOrderPaymentType obj = static_cast<OrdersStatsOrderPaymentType> (getPaymentType());
		GError *mygerror;
		mygerror = NULL;
		node = json_from_string(obj.toJson(), &mygerror);
		
	}
	const gchar *paymentTypeKey = "paymentType";
	json_object_set_member(pJsonObject, paymentTypeKey, node);
	if (isprimitive("bool")) {
		bool obj = getFake();
		node = converttoJson(&obj, "bool", "");
	}
	else {
		
	}
	const gchar *fakeKey = "fake";
	json_object_set_member(pJsonObject, fakeKey, node);
	if (isprimitive("OrdersStatsDeliveryRegionDTO")) {
		OrdersStatsDeliveryRegionDTO obj = getDeliveryRegion();
		node = converttoJson(&obj, "OrdersStatsDeliveryRegionDTO", "");
	}
	else {
		
		OrdersStatsDeliveryRegionDTO obj = static_cast<OrdersStatsDeliveryRegionDTO> (getDeliveryRegion());
		GError *mygerror;
		mygerror = NULL;
		node = json_from_string(obj.toJson(), &mygerror);
		
	}
	const gchar *deliveryRegionKey = "deliveryRegion";
	json_object_set_member(pJsonObject, deliveryRegionKey, node);
	if (isprimitive("OrdersStatsItemDTO")) {
		list<OrdersStatsItemDTO> new_list = static_cast<list <OrdersStatsItemDTO> > (getItems());
		node = converttoJson(&new_list, "OrdersStatsItemDTO", "array");
	} else {
		node = json_node_alloc();
		list<OrdersStatsItemDTO> new_list = static_cast<list <OrdersStatsItemDTO> > (getItems());
		JsonArray* json_array = json_array_new();
		GError *mygerror;
		
		for (list<OrdersStatsItemDTO>::iterator it = new_list.begin(); it != new_list.end(); it++) {
			mygerror = NULL;
			OrdersStatsItemDTO obj = *it;
			JsonNode *node_temp = json_from_string(obj.toJson(), &mygerror);
			json_array_add_element(json_array, node_temp);
			g_clear_error(&mygerror);
		}
		json_node_init_array(node, json_array);
		json_array_unref(json_array);
		
	}


	
	const gchar *itemsKey = "items";
	json_object_set_member(pJsonObject, itemsKey, node);
	if (isprimitive("OrdersStatsItemDTO")) {
		list<OrdersStatsItemDTO> new_list = static_cast<list <OrdersStatsItemDTO> > (getInitialItems());
		node = converttoJson(&new_list, "OrdersStatsItemDTO", "array");
	} else {
		node = json_node_alloc();
		list<OrdersStatsItemDTO> new_list = static_cast<list <OrdersStatsItemDTO> > (getInitialItems());
		JsonArray* json_array = json_array_new();
		GError *mygerror;
		
		for (list<OrdersStatsItemDTO>::iterator it = new_list.begin(); it != new_list.end(); it++) {
			mygerror = NULL;
			OrdersStatsItemDTO obj = *it;
			JsonNode *node_temp = json_from_string(obj.toJson(), &mygerror);
			json_array_add_element(json_array, node_temp);
			g_clear_error(&mygerror);
		}
		json_node_init_array(node, json_array);
		json_array_unref(json_array);
		
	}


	
	const gchar *initialItemsKey = "initialItems";
	json_object_set_member(pJsonObject, initialItemsKey, node);
	if (isprimitive("OrdersStatsPaymentDTO")) {
		list<OrdersStatsPaymentDTO> new_list = static_cast<list <OrdersStatsPaymentDTO> > (getPayments());
		node = converttoJson(&new_list, "OrdersStatsPaymentDTO", "array");
	} else {
		node = json_node_alloc();
		list<OrdersStatsPaymentDTO> new_list = static_cast<list <OrdersStatsPaymentDTO> > (getPayments());
		JsonArray* json_array = json_array_new();
		GError *mygerror;
		
		for (list<OrdersStatsPaymentDTO>::iterator it = new_list.begin(); it != new_list.end(); it++) {
			mygerror = NULL;
			OrdersStatsPaymentDTO obj = *it;
			JsonNode *node_temp = json_from_string(obj.toJson(), &mygerror);
			json_array_add_element(json_array, node_temp);
			g_clear_error(&mygerror);
		}
		json_node_init_array(node, json_array);
		json_array_unref(json_array);
		
	}


	
	const gchar *paymentsKey = "payments";
	json_object_set_member(pJsonObject, paymentsKey, node);
	if (isprimitive("OrdersStatsCommissionDTO")) {
		list<OrdersStatsCommissionDTO> new_list = static_cast<list <OrdersStatsCommissionDTO> > (getCommissions());
		node = converttoJson(&new_list, "OrdersStatsCommissionDTO", "array");
	} else {
		node = json_node_alloc();
		list<OrdersStatsCommissionDTO> new_list = static_cast<list <OrdersStatsCommissionDTO> > (getCommissions());
		JsonArray* json_array = json_array_new();
		GError *mygerror;
		
		for (list<OrdersStatsCommissionDTO>::iterator it = new_list.begin(); it != new_list.end(); it++) {
			mygerror = NULL;
			OrdersStatsCommissionDTO obj = *it;
			JsonNode *node_temp = json_from_string(obj.toJson(), &mygerror);
			json_array_add_element(json_array, node_temp);
			g_clear_error(&mygerror);
		}
		json_node_init_array(node, json_array);
		json_array_unref(json_array);
		
	}


	
	const gchar *commissionsKey = "commissions";
	json_object_set_member(pJsonObject, commissionsKey, node);
	node = json_node_alloc();
	json_node_init(node, JSON_NODE_OBJECT);
	json_node_take_object(node, pJsonObject);
	char * ret = json_to_string(node, false);
	json_node_free(node);
	return ret;
}

long long
OrdersStatsOrderDTO::getId()
{
	return id;
}

void
OrdersStatsOrderDTO::setId(long long  id)
{
	this->id = id;
}

Date
OrdersStatsOrderDTO::getCreationDate()
{
	return creationDate;
}

void
OrdersStatsOrderDTO::setCreationDate(Date  creationDate)
{
	this->creationDate = creationDate;
}

std::string
OrdersStatsOrderDTO::getStatusUpdateDate()
{
	return statusUpdateDate;
}

void
OrdersStatsOrderDTO::setStatusUpdateDate(std::string  statusUpdateDate)
{
	this->statusUpdateDate = statusUpdateDate;
}

OrderStatsStatusType
OrdersStatsOrderDTO::getStatus()
{
	return status;
}

void
OrdersStatsOrderDTO::setStatus(OrderStatsStatusType  status)
{
	this->status = status;
}

std::string
OrdersStatsOrderDTO::getPartnerOrderId()
{
	return partnerOrderId;
}

void
OrdersStatsOrderDTO::setPartnerOrderId(std::string  partnerOrderId)
{
	this->partnerOrderId = partnerOrderId;
}

OrdersStatsOrderPaymentType
OrdersStatsOrderDTO::getPaymentType()
{
	return paymentType;
}

void
OrdersStatsOrderDTO::setPaymentType(OrdersStatsOrderPaymentType  paymentType)
{
	this->paymentType = paymentType;
}

bool
OrdersStatsOrderDTO::getFake()
{
	return fake;
}

void
OrdersStatsOrderDTO::setFake(bool  fake)
{
	this->fake = fake;
}

OrdersStatsDeliveryRegionDTO
OrdersStatsOrderDTO::getDeliveryRegion()
{
	return deliveryRegion;
}

void
OrdersStatsOrderDTO::setDeliveryRegion(OrdersStatsDeliveryRegionDTO  deliveryRegion)
{
	this->deliveryRegion = deliveryRegion;
}

std::list<OrdersStatsItemDTO>
OrdersStatsOrderDTO::getItems()
{
	return items;
}

void
OrdersStatsOrderDTO::setItems(std::list <OrdersStatsItemDTO> items)
{
	this->items = items;
}

std::list<OrdersStatsItemDTO>
OrdersStatsOrderDTO::getInitialItems()
{
	return initialItems;
}

void
OrdersStatsOrderDTO::setInitialItems(std::list <OrdersStatsItemDTO> initialItems)
{
	this->initialItems = initialItems;
}

std::list<OrdersStatsPaymentDTO>
OrdersStatsOrderDTO::getPayments()
{
	return payments;
}

void
OrdersStatsOrderDTO::setPayments(std::list <OrdersStatsPaymentDTO> payments)
{
	this->payments = payments;
}

std::list<OrdersStatsCommissionDTO>
OrdersStatsOrderDTO::getCommissions()
{
	return commissions;
}

void
OrdersStatsOrderDTO::setCommissions(std::list <OrdersStatsCommissionDTO> commissions)
{
	this->commissions = commissions;
}

File: src/main/java/org/openapitools/server/api/model/OrdersStatsOrderDTO.java
package org.openapitools.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import org.openapitools.jackson.nullable.JsonNullable;
import org.openapitools.server.api.model.OrderStatsStatusType;
import org.openapitools.server.api.model.OrdersStatsCommissionDTO;
import org.openapitools.server.api.model.OrdersStatsDeliveryRegionDTO;
import org.openapitools.server.api.model.OrdersStatsItemDTO;
import org.openapitools.server.api.model.OrdersStatsOrderPaymentType;
import org.openapitools.server.api.model.OrdersStatsPaymentDTO;

/**
 * Информация о заказе.
 **/
@JsonInclude(JsonInclude.Include.NON_NULL)
public class OrdersStatsOrderDTO   {
  
  private Long id;
  private LocalDate creationDate;
  private OffsetDateTime statusUpdateDate;
  private OrderStatsStatusType status;
  private String partnerOrderId;
  private OrdersStatsOrderPaymentType paymentType;
  private Boolean fake;
  private OrdersStatsDeliveryRegionDTO deliveryRegion;
  private List<OrdersStatsItemDTO> items = new ArrayList<>();
  private List<OrdersStatsItemDTO> initialItems;
  private List<OrdersStatsPaymentDTO> payments = new ArrayList<>();
  private List<OrdersStatsCommissionDTO> commissions = new ArrayList<>();

  public OrdersStatsOrderDTO () {

  }

  public OrdersStatsOrderDTO (Long id, LocalDate creationDate, OffsetDateTime statusUpdateDate, OrderStatsStatusType status, String partnerOrderId, OrdersStatsOrderPaymentType paymentType, Boolean fake, OrdersStatsDeliveryRegionDTO deliveryRegion, List<OrdersStatsItemDTO> items, List<OrdersStatsItemDTO> initialItems, List<OrdersStatsPaymentDTO> payments, List<OrdersStatsCommissionDTO> commissions) {
    this.id = id;
    this.creationDate = creationDate;
    this.statusUpdateDate = statusUpdateDate;
    this.status = status;
    this.partnerOrderId = partnerOrderId;
    this.paymentType = paymentType;
    this.fake = fake;
    this.deliveryRegion = deliveryRegion;
    this.items = items;
    this.initialItems = initialItems;
    this.payments = payments;
    this.commissions = commissions;
  }

    
  @JsonProperty("id")
  public Long getId() {
    return id;
  }
  public void setId(Long id) {
    this.id = id;
  }

    
  @JsonProperty("creationDate")
  public LocalDate getCreationDate() {
    return creationDate;
  }
  public void setCreationDate(LocalDate creationDate) {
    this.creationDate = creationDate;
  }

    
  @JsonProperty("statusUpdateDate")
  public OffsetDateTime getStatusUpdateDate() {
    return statusUpdateDate;
  }
  public void setStatusUpdateDate(OffsetDateTime statusUpdateDate) {
    this.statusUpdateDate = statusUpdateDate;
  }

    
  @JsonProperty("status")
  public OrderStatsStatusType getStatus() {
    return status;
  }
  public void setStatus(OrderStatsStatusType status) {
    this.status = status;
  }

    
  @JsonProperty("partnerOrderId")
  public String getPartnerOrderId() {
    return partnerOrderId;
  }
  public void setPartnerOrderId(String partnerOrderId) {
    this.partnerOrderId = partnerOrderId;
  }

    
  @JsonProperty("paymentType")
  public OrdersStatsOrderPaymentType getPaymentType() {
    return paymentType;
  }
  public void setPaymentType(OrdersStatsOrderPaymentType paymentType) {
    this.paymentType = paymentType;
  }

    
  @JsonProperty("fake")
  public Boolean getFake() {
    return fake;
  }
  public void setFake(Boolean fake) {
    this.fake = fake;
  }

    
  @JsonProperty("deliveryRegion")
  public OrdersStatsDeliveryRegionDTO getDeliveryRegion() {
    return deliveryRegion;
  }
  public void setDeliveryRegion(OrdersStatsDeliveryRegionDTO deliveryRegion) {
    this.deliveryRegion = deliveryRegion;
  }

    
  @JsonProperty("items")
  public List<OrdersStatsItemDTO> getItems() {
    return items;
  }
  public void setItems(List<OrdersStatsItemDTO> items) {
    this.items = items;
  }

    
  @JsonProperty("initialItems")
  public List<OrdersStatsItemDTO> getInitialItems() {
    return initialItems;
  }
  public void setInitialItems(List<OrdersStatsItemDTO> initialItems) {
    this.initialItems = initialItems;
  }

    
  @JsonProperty("payments")
  public List<OrdersStatsPaymentDTO> getPayments() {
    return payments;
  }
  public void setPayments(List<OrdersStatsPaymentDTO> payments) {
    this.payments = payments;
  }

    
  @JsonProperty("commissions")
  public List<OrdersStatsCommissionDTO> getCommissions() {
    return commissions;
  }
  public void setCommissions(List<OrdersStatsCommissionDTO> commissions) {
    this.commissions = commissions;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    OrdersStatsOrderDTO ordersStatsOrderDTO = (OrdersStatsOrderDTO) o;
    return Objects.equals(id, ordersStatsOrderDTO.id) &&
        Objects.equals(creationDate, ordersStatsOrderDTO.creationDate) &&
        Objects.equals(statusUpdateDate, ordersStatsOrderDTO.statusUpdateDate) &&
        Objects.equals(status, ordersStatsOrderDTO.status) &&
        Objects.equals(partnerOrderId, ordersStatsOrderDTO.partnerOrderId) &&
        Objects.equals(paymentType, ordersStatsOrderDTO.paymentType) &&
        Objects.equals(fake, ordersStatsOrderDTO.fake) &&
        Objects.equals(deliveryRegion, ordersStatsOrderDTO.deliveryRegion) &&
        Objects.equals(items, ordersStatsOrderDTO.items) &&
        Objects.equals(initialItems, ordersStatsOrderDTO.initialItems) &&
        Objects.equals(payments, ordersStatsOrderDTO.payments) &&
        Objects.equals(commissions, ordersStatsOrderDTO.commissions);
  }

  @Override
  public int hashCode() {
    return Objects.hash(id, creationDate, statusUpdateDate, status, partnerOrderId, paymentType, fake, deliveryRegion, items, initialItems, payments, commissions);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class OrdersStatsOrderDTO {\n");
    
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    creationDate: ").append(toIndentedString(creationDate)).append("\n");
    sb.append("    statusUpdateDate: ").append(toIndentedString(statusUpdateDate)).append("\n");
    sb.append("    status: ").append(toIndentedString(status)).append("\n");
    sb.append("    partnerOrderId: ").append(toIndentedString(partnerOrderId)).append("\n");
    sb.append("    paymentType: ").append(toIndentedString(paymentType)).append("\n");
    sb.append("    fake: ").append(toIndentedString(fake)).append("\n");
    sb.append("    deliveryRegion: ").append(toIndentedString(deliveryRegion)).append("\n");
    sb.append("    items: ").append(toIndentedString(items)).append("\n");
    sb.append("    initialItems: ").append(toIndentedString(initialItems)).append("\n");
    sb.append("    payments: ").append(toIndentedString(payments)).append("\n");
    sb.append("    commissions: ").append(toIndentedString(commissions)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
